src/DynamoCoreWpf/UI/Converters.cs

```diff
@@ -396,17 +396,19 @@ public object ConvertBack(object value, Type targetType, object parameter, Syste
         }
     }
 
-    public class PathToFileNameConverter : IValueConverter
+    public class WorkspaceToFileNameConverter : IValueConverter
     {
         public object Convert(object value, Type targetType, object parameter, System.Globalization.CultureInfo culture)
         {
-            if (value is string && !string.IsNullOrEmpty(value as string))
+
+            var wsvm = value as WorkspaceViewModel;
+            if (wsvm != null && !string.IsNullOrEmpty(wsvm.FileName))
             {
                 // Convert to path, get file name. If read-only file, append [Read-Only].
-                if (DynamoUtilities.PathHelper.IsReadOnlyPath((string)value))
-                    return Resources.TabFileNameReadOnlyPrefix + Path.GetFileName((string)value);
+                if (wsvm.Model.IsReadOnly)
```

Comment:
OK, so it is not displaying `Read-only` before because it is actually checking if the file path is read only but not the workspace, before it was equal but no longer ture.  I see now